src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,265 @@
+        superClass, elements, tree.annotations, tree.typeParameters);
+
+    if (initStaticVars.length === 0) {
+      return classExpression;
+    }
+
+    this.pushTempScope();
+    let id = this.getTempIdentifier();
+    let className = createBindingIdentifier(id);
+    let statements = [
+      parseStatement `let ${id} = ${classExpression}`,
+      ...createStaticInitializerStatements(className, initStaticVars),
+      parseStatement `return ${className}`
+    ];
+    let body = createFunctionBody(statements);
+    this.popTempScope();
+
+    return createImmediatelyInvokedFunctionExpression(body);
+  }
+
+  getDefaultConstructor_(tree) {
+    if (tree.superClass) {
+      return parsePropertyDefinition `constructor() {
+        super(...arguments);
+      }`;
+    } else {
+      return parsePropertyDefinition `constructor() {}`;
+    }
+  }
+
+  appendInstanceInitializers_(constructor, initExpression, superClass) {
+    if (superClass) {
+      let transformer = new SuperExpressionTransformer(initExpression);
+      return transformer.transformAny(constructor);
+    }
+
+    let statements = constructor.body.statements;
+    let initStatement = createExpressionStatement(initExpression);
+    statements = prependStatements(statements, initStatement);
+
+    return new PropertyMethodAssignment(constructor.location, false,
+        constructor.functionKind, constructor.name, constructor.parameterList,
+        constructor.typeAnnotation, constructor.annotations,
+        createFunctionBody(statements));
+  }
+}
+
+// TODO(vicb): Does not handle computed properties
+function createStaticInitializerStatements(className, initStaticMemberVars) {
+  let statements = [];
+  for (let i = 0; i < initStaticMemberVars.length; i++) {
+    let mv = initStaticMemberVars[i];
+    let propName = mv.name.literalToken.value;
+    let statement = parseStatement
+        `Object.defineProperty(${className}, ${propName}, {value: ${mv.initializer},
+        enumerable: true, configurable: true, writable: true})`;
+    statements.push(statement);
+  }
+  return statements;
+}
+
+// TODO(vicb): Does not handle computed properties
+function getInstanceInitExpression(initInstanceVars) {
+    let expressions = [];
+    for (let i = 0; i < initInstanceVars.length; i++) {
+      let mv = initInstanceVars[i];
+      let name = mv.name.literalToken;
+      expressions.push(parseExpression `this.${name} = ${mv.initializer}`);
+    }
+  return createCommaExpression(expressions);
+}
+
+class SuperExpressionTransformer extends ParseTreeTransformer {
+  constructor(expression) {
+    super();
+    this.expression = expression;
+  }
+
+  transformCallExpression(tree) {
+    if (tree.operand.type === SUPER_EXPRESSION) {
+      var thisExpression = parseExpression `this`;
+      return createParenExpression(
+          createCommaExpression([tree, this.expression, thisExpression]));
+    }
+
+    return tree;
+  }
```

Comment:
I'm little bit concerned about nested classes:

```js
class A extends B {
  x = 42;
  constructor() {
    super();
    class C extends D {
      constructor() {
        super(); // Ooops.
      }
  }
}
```


I think the simplest solution is just to add:

transformClassDeclaration(tree) { return tree; }
transformClassExpression(tree) { return tree; }